### DISCUSSION TITLE
Member avatar outdated

### DISCUSSION THREAD

**Discussion by iHazzu** [Rapptz/discord.py#6034] [Q&A] [answered]
I have a bot where I need to redeem the members avatar in a command. For this I use `await guild.fetch_member(member_id)`. However, if the user changes the avatar after the bot initiates its connection with the discord, `avatar_url` is not updated, even if I use `fetch_member` again. Is there a way to retrieve avatars correctly without having to cache members?
Here is the bot object:
```
bot = commands.Bot(
    command_prefix = db.prefix,
    case_insensitive = True,
    help_command = None,
    intents = Intents(messages=True, reactions=True, guilds=True, emojis=True)
)
```

**Comment by Rapptz** [comment, marked as answer]
This is due to the global user cache and is a bad interaction with it. There's no good way of fixing this since it's a library bug that needs to be fixed. One way to bandage it at the moment would be to disable all forms of member caching explicitly by setting a `MemberCacheFlags`

**Reply by iHazzu** [reply]
I tried this, but the problem persists 😿 
```
bot = commands.Bot(
    command_prefix = db.prefix,
    case_insensitive = True,
    help_command = None,
    intents = Intents(messages=True, reactions=True, guilds=True, emojis=True),
    member_cache_flags = MemberCacheFlags.none()
)
```

**Reply by Harmon758** [reply]
I think this should be fixed with https://github.com/Rapptz/discord.py/commit/7a3a571e0a641aaeb061e86d8a6e851c5cd4c381 in the latest development version on the master branch.
